executor/instruction/has_reverse_executor.rs

```diff
@@ -109,25 +106,30 @@ impl HasReverseExecutor {
             owner_types,
             filter_fn,
             attribute_cache,
+            checker,
         })
     }
 
-    pub(crate) fn get_iterator<Snapshot: ReadableSnapshot>(
+    pub(crate) fn get_iterator(
         &self,
-        snapshot: &Snapshot,
-        thing_manager: &ThingManager,
+        snapshot: &Arc<impl ReadableSnapshot + 'static>,
```

Comment:
Do you want to just take the `Arc` directly? You're going to have to clone it into the checker anyway right?

Reply:
No, only checks that involve edges need to clone the arc.